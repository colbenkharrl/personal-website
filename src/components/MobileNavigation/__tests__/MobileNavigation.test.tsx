import { fireEvent, getByTestId, render, screen } from '@testing-library/react';
import { MobileNavigation } from '../';

// mock useRouter
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
  usePathname: jest.fn(),
}));
jest.mock('../../MobileNavItem', () => ({
  MobileNavItem: () => <p>MobileNavItem</p>,
}));

it('MobileNavigation matches snapshot while closed (regression test)', async () => {
  render(<MobileNavigation data-testid="MobileNavigation" />);

  expect(screen.getByTestId('MobileNavigation')).toMatchSnapshot(
    'Full component snapshot.',
  );
});

it('MobileNavigation matches snapshot while opened (regression test)', async () => {
  render(<MobileNavigation data-testid="MobileNavigation" />);

  fireEvent.click(screen.getByTestId('MobileNavigation-button'));

  expect(screen.getByTestId('MobileNavigation')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
