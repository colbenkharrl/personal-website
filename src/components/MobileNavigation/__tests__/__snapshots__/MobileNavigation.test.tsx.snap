// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation matches snapshot while closed (regression test): Full component snapshot. 1`] = `
<div
  data-headlessui-state=""
  data-testid="MobileNavigation"
>
  <button
    aria-expanded="false"
    class="group flex items-center rounded-full bg-white/90 px-4 py-2 text-sm font-medium text-zinc-800 shadow-lg shadow-zinc-800/5 ring-1 ring-zinc-900/5 backdrop-blur dark:bg-zinc-800/90 dark:text-zinc-200 dark:ring-white/10 dark:hover:ring-white/20"
    data-headlessui-state=""
    data-testid="MobileNavigation-button"
    id="headlessui-popover-button-:r0:"
    type="button"
  >
    Menu
    <svg
      aria-hidden="true"
      class="ml-3 h-auto w-2 stroke-zinc-500 group-hover:stroke-zinc-700 dark:group-hover:stroke-zinc-400"
      viewBox="0 0 8 6"
    >
      <path
        d="M1.75 1.75 4 4.25l2.25-2.5"
        fill="none"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </button>
</div>
`;

exports[`MobileNavigation matches snapshot while opened (regression test): Full component snapshot. 1`] = `
<div
  data-headlessui-state="open"
  data-testid="MobileNavigation"
>
  <button
    aria-controls="headlessui-popover-panel-:r5:"
    aria-expanded="true"
    class="group flex items-center rounded-full bg-white/90 px-4 py-2 text-sm font-medium text-zinc-800 shadow-lg shadow-zinc-800/5 ring-1 ring-zinc-900/5 backdrop-blur dark:bg-zinc-800/90 dark:text-zinc-200 dark:ring-white/10 dark:hover:ring-white/20"
    data-headlessui-state="open"
    data-testid="MobileNavigation-button"
    id="headlessui-popover-button-:r2:"
    type="button"
  >
    Menu
    <svg
      aria-hidden="true"
      class="ml-3 h-auto w-2 stroke-zinc-500 group-hover:stroke-zinc-700 dark:group-hover:stroke-zinc-400"
      viewBox="0 0 8 6"
    >
      <path
        d="M1.75 1.75 4 4.25l2.25-2.5"
        fill="none"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </button>
  <div>
    <div
      aria-hidden="true"
      class="fixed inset-0 z-50 bg-zinc-800/40 backdrop-blur-sm dark:bg-black/80 duration-150 ease-out opacity-0"
      data-headlessui-state="open"
      id="headlessui-popover-overlay-:r4:"
    />
    <div
      class="fixed inset-x-4 top-8 z-50 origin-top rounded-3xl bg-white p-8 ring-1 ring-zinc-900/5 dark:bg-zinc-900 dark:ring-zinc-800 duration-150 ease-out opacity-0 scale-95"
      data-headlessui-state="open"
      id="headlessui-popover-panel-:r5:"
      tabindex="-1"
    >
      <div
        class="flex flex-row-reverse items-center justify-between"
      >
        <button
          aria-label="Close menu"
          class="-m-1 p-1"
          data-headlessui-state="open"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="h-6 w-6 text-zinc-500 dark:text-zinc-400"
            viewBox="0 0 24 24"
          >
            <path
              d="m17.25 6.75-10.5 10.5M6.75 6.75l10.5 10.5"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </button>
        <h2
          class="text-sm font-medium text-zinc-600 dark:text-zinc-400"
        >
          Navigation
        </h2>
      </div>
      <nav
        class="mt-6"
      >
        <ul
          class="-my-2 divide-y divide-zinc-100 text-base text-zinc-800 dark:divide-zinc-100/5 dark:text-zinc-300"
        >
          <p>
            MobileNavItem
          </p>
          <p>
            MobileNavItem
          </p>
          <p>
            MobileNavItem
          </p>
          <p>
            MobileNavItem
          </p>
          <p>
            MobileNavItem
          </p>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
